daemon/container_operations_unix.go

```diff
@@ -618,16 +618,39 @@ func (daemon *Daemon) allocateNetwork(container *container.Container) error {
 	}
 
 	updateSettings := false
+	if container.Config.NetworkDisabled || container.HostConfig.NetworkMode.IsContainer() {
+		return nil
+	}
 	if len(container.NetworkSettings.Networks) == 0 {
-		if container.Config.NetworkDisabled || container.HostConfig.NetworkMode.IsContainer() {
-			return nil
-		}
-
 		err := daemon.updateContainerNetworkSettings(container, nil)
 		if err != nil {
 			return err
 		}
 		updateSettings = true
+	} else {
+		networkName := container.HostConfig.NetworkMode.NetworkName()
+		if container.HostConfig.NetworkMode.IsDefault() {
+			networkName = daemon.netController.Config().Daemon.DefaultNetwork
+		}
+		if container.HostConfig.NetworkMode.IsUserDefined() {
+			n, err := daemon.FindNetwork(networkName)
+			if err != nil {
+				return err
+			}
+			networkName = n.Name()
+		}
+		if _, ok := container.NetworkSettings.Networks[networkName]; !ok {
```

Comment:
So it'll break if the first network that the container was connected to, is disconnected?